perl/lib/Seq/Output.pm

```diff
@@ -61,8 +70,24 @@ sub BUILD {
   }
 }
 
-# TODO: will be singleton, configured once for all consumers
-sub initialize {
+sub uniqueify {
```

Comment:
seems worth a test or two just to verify that the tricky bits like checking for undefined values, converting `keys %count` to scalar are working as intended.

(Technically this sub could return `$_[0]` as soon as we detect two distinct values, but unless it's in a tight inner loop and the input array is typically large the perf probably isn't worth the additional complexity)

Reply:
Yeah I didn't micro bench, branching in loops can be cache unfriendly and most of the arrays it will checking are very small (<3 length I would guess)